src/integer_mod_q/mat_zq/serialize.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2023 Marvin Beckmann
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains implementations of functions
+//! important for serialization such as the [`Serialize`] and [`Deserialize`] trait.
+//!
+//! The explicit functions contain the documentation.
+
+use super::MatZq;
+use crate::macros::serialize::{deserialize, serialize};
+use core::fmt;
+use serde::{
+    de::{Error, MapAccess, Unexpected, Visitor},
+    ser::SerializeStruct,
+    Deserialize, Serialize,
+};
+use std::str::FromStr;
+
+serialize!("matrix", MatZq);
+deserialize!("matrix", Matrix, MatZq);
+
+#[cfg(test)]
+mod test_serialize {
+    use crate::integer_mod_q::MatZq;
+    use std::str::FromStr;
+
+    /// tests whether the serialization of a positive [`MatZq`] works.
+    #[test]
+    fn serialize_output_positive() {
+        let mat_poly_str = "[[17, 42],[1, 17]] mod 57";
+        let mat_poly_z = MatZq::from_str(mat_poly_str).unwrap();
+        let cmp_string = format!("{{\"matrix\":\"{}\"}}", mat_poly_str);
+
+        assert_eq!(cmp_string, serde_json::to_string(&mat_poly_z).unwrap())
+    }
+
+    /// tests whether the serialization of a negative [`MatZq`] works.
+    #[test]
+    fn serialize_output_negative() {
+        let mat_poly_str = "[[-17, -42, 1],[-13, -5, -42]] mod 57";
+        let mat_poly_z = MatZq::from_str(mat_poly_str).unwrap();
+        let cmp_string = "{\"matrix\":\"[[40, 15, 1],[44, 52, 15]] mod 57\"}";
+
+        assert_eq!(cmp_string, serde_json::to_string(&mat_poly_z).unwrap())
+    }
+
+    /// tests whether the serialization of a positive large [`MatZq`] works.
+    #[test]
+    fn serialize_output_positive_large() {
+        let mat_poly_str = format!("[[3, 17, {}, 1, 2, 13, 5]] mod {}", u64::MAX - 1, u64::MAX);
+        let mat_poly_z = MatZq::from_str(&mat_poly_str).unwrap();
+        let cmp_string = format!("{{\"matrix\":\"{}\"}}", mat_poly_str);
+
+        assert_eq!(cmp_string, serde_json::to_string(&mat_poly_z).unwrap())
+    }
```

Comment:
Please add a test for negative large values for this case as you did in all other cases, too. (Just to stay consistent along these functions.

Reply:
ahh, saw it too late, will make a PR for it...